### PR TITLE
create docs tools. Implement IFRAME tool support

### PR DESCRIPTION
**Pull request by ryanmelt**

### CODE REVIEW

File: docs.openc3.com/plugin.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+TOOL docs "Documentation"
+  URL /tools/docs/index.html
+  INLINE_URL nil
+  ICON mdi-hexadecimal
```
Comment (jmthomas) [inline docs.openc3.com/plugin.txt:4]: I think this icon should be `mdi-file-document-multiple` but we already use that for handbooks. So for handbooks maybe `view-list-outline` or `list-box` (https://pictogrammers.com/library/mdi/)?

File: docs.openc3.com/plugin.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  DISABLE_ERB
+  WINDOW IFRAME
+  POSITION 15
```
Comment (jmthomas) [inline docs.openc3.com/plugin.txt:7]: Should this go in `CATEGORY Utilities`?

**Review by jmthomas — CHANGES_REQUESTED** [review]

**Comment by jmthomas** [comment]
Overall this is awesome as a tool. Love the way you made it build either for the site itself or as a tool. Keeping it DRY!

File: compose-build.yaml [hunk 3]
```diff
@@ -117,6 +117,8 @@ services:
   openc3-cosmos-init:
     build:
       context: "openc3-cosmos-init"
+      additional_contexts:
+        docs: "docs.openc3.com"
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply compose-build.yaml:121]: It adds in the docs.openc3.com folder to the docker build context.  Needed to build the tool which is in a different path than all the other tools.

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-iframe/src/tools/Iframe/Iframe.vue [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      "
+    >
+      <iframe
+        :src="url"
+        style="
+          flex-grow: 1;
+          border: none;
+          margin: 0;
+          padding: 0;
+          height: calc(100vh - 118px);
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply openc3-cosmos-init/plugins/packages/openc3-cosmos-tool-iframe/src/tools/Iframe/Iframe.vue:39]: Measured from the size of the header and footer plus some padding.

File: openc3/lib/openc3/models/tool_model.rb [hunk 5]
```diff
@@ -152,16 +154,19 @@ def initialize(
...
 
     def create(update: false, force: false)
       tools = self.class.all(scope: @scope)
 
       # Make sure a tool with this folder_name doesn't already exist
       unless update
-        tools.each do |_tool_name, tool|
-          if tool['folder_name'] == @folder_name
-            raise "Tool with folder_name #{@folder_name} already exists at create"
+        if @folder_name
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply openc3/lib/openc3/models/tool_model.rb:165]: This is to support multiple tools with no folder_name (like plain links to Google or whatever).  Before this logic was limiting to only one.

File: docs.openc3.com/plugin.txt (hunk 2 above)
Comment (ryanmelt) [reply docs.openc3.com/plugin.txt:7]: No I like it separate at the bottom.

File: docs.openc3.com/plugin.txt (hunk 1 above)
Comment (ryanmelt) [reply docs.openc3.com/plugin.txt]: changed to mdi-bookshelf

**Pull request merged**